### PR TITLE
feat: add mfa client library binding(deactivated by default)

### PR DESCRIPTION
**Pull request by J0**
Introduces the additional MFA methods for admin and regular endpoints. 

Admin:
- `deleteFactor`
- `listFactors`

Regular:
- `verify`
- `enroll`
- `unenroll`
- `challenge`
-  `listFactors`
-  `getAuthenticatorAssuranceLevel`

Pull Requests involved:
- #440
- #446
- #472
- #475
- #478
- #479
- #480
- #486
- #506
- #508
- #510

### CODE REVIEW

File: src/GoTrueAdminApi.ts [hunk 1]
```diff
@@ -227,4 +235,8 @@ export default class GoTrueAdminApi {
       throw error
     }
   }
+
+  async _deleteFactor() {
```
Comment (hf) [inline src/GoTrueAdminApi.ts:239]: Mark it as `private` to be hidden from TypeScript users? (Note there's no such thing as "private" in JavaScript, it's only enforced by TypeScript, thus the `_` prefix.)

File: src/GoTrueAdminApi.ts [hunk 2]
```diff
@@ -213,7 +223,7 @@ export default class GoTrueAdminApi {
    *
    * This function should only be called on a server. Never expose your `service_role` key in the browser.
    */
-  async deleteUser(uid: string): Promise<UserResponse> {
+  private async deleteUser(uid: string): Promise<UserResponse> {
```
Comment (kangmingtay) [inline src/GoTrueAdminApi.ts:226]: hmm why is this private?

Comment (J0) [reply src/GoTrueAdminApi.ts:226]: Yup, good catch — It’s not meant to be private—  was testing something and forgot to change back. Let me also delete the unused admin functions

**Comment by kangmingtay** [comment]
@J0 hmm can we omit the docs stuff and keep that in a separate PR?

**Review by hf — APPROVED** [review]
🚀

**Pull request merged**